quipucords/api/report/renderer.py

```diff
@@ -138,19 +147,51 @@ def render(self,
         if not systems_list:
             return None
         csv_writer.writerow(['Report:'])
-
         headers = csv_helper.generate_headers(
-            systems_list, exclude=set([
-                'id', 'report_id', 'metadata', 'entitlements']))
+            systems_list,
+            exclude={'id', 'report_id', 'metadata', 'entitlements'})
+        if self.SOURCES_KEY in headers:
+            headers += self.source_headers
+            headers = sorted(list(set(headers)))
+
+        # Add source heaaders
         csv_writer.writerow(headers)
         for system in systems_list:
             row = []
+            system_sources = system.get(self.SOURCES_KEY)
+            if system_sources is not None:
+                sources_info = self._compute_source_info(system_sources)
+            else:
+                sources_info = None
             for header in headers:
-                fact_value = system.get(header)
+                fact_value = None
+                if header in self.source_headers:
+                    if sources_info is not None:
+                        fact_value = sources_info.get(header)
```

Comment:
Is there a case here where fact value doesn't get set if source_info is None? Is None a good default or is empty string (just want to make sure we have the appropriate commas being generated)

Reply:
None is good. Leaves column blank.

Reply:
ok